spicy/toolchain/src/compiler/parser/parser.yy

```diff
@@ -884,7 +884,7 @@ unit_hook_id: { driver->enableHookIDMode(); }
 
 unit_hook_attribute
               : FOREACH                          { $$ = builder->attribute("foreach", __loc__); }
-              | PRIORITY '=' expr                { $$ = builder->attribute("priority", std::move($3), __loc__); }
+              | PRIORITY '=' expr                { $$ = builder->attribute("&priority", std::move($3), __loc__); }
```

Comment:
This makes sense for internal consistency. However, even previously we were seemingly adding `&priority` during codegen time; see https://github.com/zeek/spicy/blob/1825bdbaea723ab184b45687821e9bee137d8d65/spicy/toolchain/src/compiler/codegen/codegen.cc#L587.  Was that not working, and/or is now no longer needed?

Reply:
_I believe_ this is still needed. The issue fixed here is about a previously incorrectly persisted hook `priority` from Spicy user code which was set in a way that `Hook::priority` would not recognize,

https://github.com/zeek/spicy/blob/d33f06f9c44e20beeae695b1c29973396da288ca/spicy/toolchain/include/ast/declarations/hook.h#L51-L56

We use `Hook::priority` as input to `compileHook` so this is now consistent. I am not sure, but it could be the flow is more complicated than needed, but I didn't touch this in this PR.

Reply:
> I'll add a patch deprecating `priority` for `&priority` in Spicy.

Thinking about this more, switching this to attribute syntax could potentially allow surprising syntax (especially if we clean up the parser to use generic attribute parsing), e.g.,

```ruby
: bytes &eod &priority=4711 {}
```

which we could even allow to parse from

```ruby
: bytes &priority=4711 &eod {}
```

Allowing to blur what is a parsing attribute and what a hook attribute doesn't seem ideal.

In Spicy we already allow `%debug` as a hook property (stored internally as an attribute). What do you think about making syntax property syntax for `%priority` as well? (We also have a `foreach` attribute, but that syntax without `&` or `%` seems useful to me).

Reply:
I'll punt on deprecating `priority`, and will make the cleanup in the lexer.